src/coreclr/src/gc/unix/cgroup.cpp

```diff
@@ -348,6 +375,50 @@ class CGroup
         return cgroup_path;
     }
 
+    static bool GetCGroupMemoryLimit(uint64_t *val, const char *filename)
+    {
+        bool result = false;
+
+        if (s_memory_cgroup_path == nullptr)
+            return result;
+
+        char* mem_limit_filename = nullptr;
+        if (asprintf(&mem_limit_filename, "%s%s", s_memory_cgroup_path, filename) < 0)
+            return result;
+
+        result = ReadMemoryValueFromFile(mem_limit_filename, val);
+        free(mem_limit_filename);
+        return result;
+    }
+
+    static bool GetCGroupMemoryUsage(size_t *val, const char *filename)
+    {
+        bool result = false;
+
+        if (s_memory_cgroup_path == nullptr)
+            return result;
+
+        char* mem_usage_filename = nullptr;
+        if (asprintf(&mem_usage_filename, "%s%s", s_memory_cgroup_path, filename) < 0)
+            return result;
+
+        uint64_t temp = 0;
+        result = ReadMemoryValueFromFile(mem_usage_filename, &temp);
+        if (result)
+        {
+            if (temp > std::numeric_limits<size_t>::max())
```

Comment:
This is always going to be false on a 64-bit system. Do we force `size_t` to be 64-bit on 32-bit systems?

Reply:
I am not sure. This code was added as part of https://github.com/dotnet/coreclr/pull/25724. I only moved it around.

Reply:
Right, that was the intent of this check.